### ISSUE TITLE
Profile all functions option

### ISSUE THREAD

**Issue by iddl** [rkern/line_profiler#45]
I'm working on an atom plugin capable of parsing output from line_profiler. At times it can be painful to explicitly mark all the functions to profile. Would it be possible to add an option to record samples from all function calls ?

**Comment by j-towns** [comment]
Check out the project I'm working on [here](https://github.com/j-towns/iprofiler). I managed to modify line_profiler to do what you're asking, by changing a few lines in _line_profiler.pyx (the modified version is in the 'line_profiler' folder in the above project).

**Comment by iddl** [comment]
@j-towns thanks a lot. That worked.
I wish this feature was included upstream, worth making a pull request ?

**Comment by Zeye1** [comment]
@j-towns  Hi Mr Townsend i have replace the file  _line_profiler.pyx of line_profiler. Then i use "kernprof -l hello2.py" and "python -m  line_profiler hello2.py.lprof", But it generate nothing in reasult.  Is there other modification ?

**Comment by Zeye1** [comment]
@iddl Hi, i have replace the file _line_profiler.pyx of line_profiler. Then i use "kernprof -l hello2.py" and "python -m line_profiler hello2.py.lprof", But it generate nothing in reasult. Is there other modification ?

**Comment by iddl** [comment]
@Zeye1 I've never attempted running the line profiler modification from `kernprof ...` or `python -m line_profiler ...` I can try that out and let you know.

I can try opening a pull request for this since a few people look interested in the feature.

**Comment by Zeye1** [comment]
@iddl  yes, but i wonder how can you profile all the functions without marking by @profile

**Comment by iddl** [comment]
@Zeye1 you could modify https://github.com/rkern/line_profiler/blob/e8cffaa11f6ca12c79edb344993126d2efebc8b3/kernprof.py#L222 to do something like `prof.runcall(execfile, script_file, ns, ns)` if you want to profile the module main and everything that gets executed with it.

Side note: I find it useful to be able to specify where I want to start my profile, I wouldn't want to add this 'profile main' option to kernprof, it would only make things confusing IMO.

**Comment by iddl** [comment]
https://github.com/rkern/line_profiler/pull/76
